strategies/changed/src/main/java/org/arquillian/smart/testing/vcs/git/GitChangesDetector.java

```diff
@@ -75,7 +93,7 @@ private File findRepoRoot(File currentDir) {
             .map(diffEntry -> {
                 try {
                     final File sourceFile = new File(repoRoot, diffEntry.getNewPath());
-                    return new ClassNameExtractor().extractFullyQualifiedName(sourceFile);
+                    return ClassNameExtractor.extractFullyQualifiedName(sourceFile);
```

Comment:
Making it static has nothing to do with SRP, and as such has nothing to do with Object Oriented programming, that said I don't really see a reason to make it static at all :) Maybe better way would be to make this component part of the owning object through composition?